content/blog/zephyr-how-to-use-psram/index.md

```diff
@@ -0,0 +1,605 @@
+
+In a simplified way, we can describe the relationship among PSRAM, the MMU, and the CPU as follows:
+
+- Data is automatically copied from PSRAM to the data cache when the CPU attempts to access a memory position not currently in the cache.
+
+- Conversely, when CPU modifies data in the cache, the updated contents are copied back to PSRAM.
+
+As of the time of writing, although ESP32, ESP32-S2, ESP32-S3, ESP32-C61, and ESP32-P4 support external PSRAM, only the former three are currently supported on Zephyr.
+
+## 2. ESP32 modules and PSRAM
+
+To know which ESP32 modules contain PSRAM we have to undestand how modules part nummber are formed. Lets use as an example the moduel whose part number is **ESP32-S3N8R8**
+
+| Field           | Contents | Meaning             |
+|:---------------:|:--------:|:-------------------:|
+| SoC series      | ESP32-S3 | ESP32-S3 SoC series |
+| FLASH size      | N8       | 8 MBytes FLASH      |
+| PSRAM size [^1] | R8       | 8 MBytes PSRAM      |
+
+ [^1]: If the module does not contain PSRAM, this field is absent
+
+
+## 3. Custom hardware and PSRAM
+
+Although Espressif offers modules featuring PSRAM across various ESP32 SoC series, there may be cases where creating custom hardware must be developed to meet specific requirements, such as PSRAM capacity or the SPI interface bus width, which impacts the time for each memory operation and overall data access speed. Additionally, there may be other reasons unrelated to PSRAM that necessitate custom hardware.
+
+In this case must be observed PSRAM will share data I/O and clock lines with external FLASH memory, but not chip-select line. PSRAM and FLASH have separated chip select lines. For information related on how to connect off-package PSRAM lines, the respective SoC datasheet must be consulted.
+
+The following SPI modes are supported to access external PSRAM memories:
+
+ - Single SPI
+ - Dual SPI
+ - Quad SPI
+ - QPI
+ - Octo SPI
+ - OPI
+
+ESP32 can manage PSRAM of capacities up 32Mbytes depending on  SoCs series.
+
+Espressif currently some PSRAM models:
+
+| Part number  | Memory Capacity | Operationg Voltage |
+|:------------:|:---------------:|:------------------:|
+| ESP-PSRAM32  | 32Mbit          | 1.8V               |
+| ESP-PSRAM32H | 32Mbit          | 3.3V               |
+| ESP-PSRAM64  | 64Mbit          | 1.8V               |
+| ESP-PSRAM64H | 64Mbit          | 3.3V               |
+
+## 4. Configuring PSRAM for use in Zephyr
+
+The Kconfig parameters related to PSRAM usage in Zephyr can be found in [zephyr/soc/espressif/commom/Kconfig.spiram](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/espressif/common/Kconfig.spiram)
+
+Below are the main parameters related to PSRAM, along with a brief description of each.
+
+### 4.1. ESP_SPIRAM
```

Comment:
To reduce the ToC, please use:

`- ESP_SPIRAM` instead of `### 4.1. ESP_SPIRAM`